static/app/components/forms/model.tsx

```diff
@@ -259,6 +268,8 @@ class FormModel {
       this.initialData[id] = props.setValue(this.initialData[id], props);
       this.fields.set(id, this.initialData[id]);
     }
+
+    this.formState = this.isFormInvalid() ? FormState.INVALID : FormState.READY;
```

Comment:
I'm not sure I understand why this function is an action and not computed?

Reply:
honestly, I'm a bit unclear on the logic. I thought we should mark the form as invalid if it’s not ready. We can definitely create a function to check the form’s validity. I’ve updated the PR - let me know if this is what you had in mind